platform/model/typical_queries/hits.go

```diff
@@ -250,6 +254,44 @@ func (query Hits) computeIdForDocument(doc model.SearchHit, defaultID string) st
 	return pseudoUniqueId
 }
 
+func ComputeHash(data json.RawMessage) string {
+	var parsed interface{}
+	if err := json.Unmarshal(data, &parsed); err != nil {
+		hash := sha256.Sum256(data)
+		return hex.EncodeToString(hash[:])
+	}
+	normalized := normalizeJSON(parsed)
+	normalizedBytes, err := json.Marshal(normalized)
+	if err != nil {
+		hash := sha256.Sum256(data)
+		return hex.EncodeToString(hash[:])
+	}
+	hash := sha256.Sum256(normalizedBytes)
+	return hex.EncodeToString(hash[:])
+}
+
+// normalizeJSON recursively normalizes JSON structure to ensure consistent ordering for further hashing.
+func normalizeJSON(v interface{}) interface{} {
```

Comment:
I do not understand how this guarantees consistent ordering. map have no guarantee of ordering. In particular entering same objects to map with a different order may result in a different map.